pkg/controllers/cspc-controller/status.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cspccontroller
+
+import (
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"github.com/openebs/api/pkg/apis/types"
+	"github.com/openebs/cstor-operators/pkg/controllers/cspc-controller/util"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/klog"
+	"time"
+)
+
+func (c *Controller) UpdateStatusEventually(cspc *cstor.CStorPoolCluster) error {
+	err := c.UpdateStatus(cspc)
+	maxRetry := 3
+
+	if err != nil {
+		klog.Errorf("failed to update cspc %s status: will retry %d times at 2s interval: {%s}",
+			cspc.Name, maxRetry, err.Error())
+		for maxRetry > 0 {
+			cspcNew, err := c.GetStoredCStorVersionClient().
+				CStorPoolClusters(cspc.Namespace).
+				Get(cspc.Name, metav1.GetOptions{})
+
+			if err != nil {
+				// this is possible due to etcd unavailability so do not retry more here
+				return errors.Wrapf(err, "failed to update cspc status")
+			}
+
+			err = c.UpdateStatus(cspcNew)
```

Comment:
here we are trying to update the cspcNew which we just got it from the etcd , i.e no changes in status

Reply:
ok  got it see below, in `UpdateStatus` func we are taking care of recalculating the differences

Reply:
```go
err= c.UpdateStatus(c.calculateStatus(cspcNew))
```
Above will help us clear understanding as the func  name suggest. ?

Reply:
The final update should recalculate.

Reply:
Abstraction helps -- client code note to worry about recalculation.